cli/cmd/plugin/ui/web/tanzu-ui/src/views/unmanaged-cluster/unmanaged-cluster-wizard-page/unmanaged-cluster-network-settings-step/UnmanagedClustersNetworkSettings.tsx

```diff
@@ -0,0 +1,172 @@
+// React imports
+import React, { ChangeEvent, useState } from 'react';
+
+// Library imports
+import { ClarityIcons, blockIcon, blocksGroupIcon, clusterIcon } from '@cds/core/icon';
+import { CdsControlMessage, CdsFormGroup } from '@cds/react/forms';
+import { CdsInput } from '@cds/react/input';
+import { useForm } from 'react-hook-form';
+import { CdsRadio, CdsRadioGroup } from '@cds/react/radio';
+import { CdsIcon } from '@cds/react/icon';
+import { CdsButton } from '@cds/react/button';
+
+// App imports
+import { INPUT_CHANGE } from '../../../../state-management/actions/Form.actions';
+import { StepProps } from '../../../../shared/components/wizard/Wizard';
+
+ClarityIcons.addIcons(blockIcon, blocksGroupIcon, clusterIcon);
+
+interface FormInputs {
+    CLUSTER_NAME: string;
+    CLUSTER_SERVICE_CIDR: string;
+    CLUSTER_POD_CIDR: string;
+}
+
+const unmanagedClusterProviders = [
+    {
+        label: 'calico',
+        value: 'CALICO',
+    },
+    {
+        label: 'anthrea',
+        value: 'ANTHREA',
+    },
+    {
+        label: 'none',
+        value: 'NONE',
+    },
+];
+function UnmanagedClusterNetworkSettings(props: Partial<StepProps>) {
+    const { handleValueChange, currentStep } = props;
+
+    const {
+        register,
+        formState: { errors },
+    } = useForm<FormInputs>();
+
+    const handleClusterNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+        if (handleValueChange) {
+            handleValueChange(INPUT_CHANGE, 'CLUSTER_NAME', event.target.value, currentStep, errors);
+        }
+    };
+
+    const handleClusterServiceCidrChange = (event: ChangeEvent<HTMLInputElement>) => {
+        if (handleValueChange) {
+            handleValueChange(INPUT_CHANGE, 'CLUSTER_SERVICE_CIDR', event.target.value, currentStep, errors);
+        }
+    };
+
+    const handleClusterPodCidrChange = (event: ChangeEvent<HTMLInputElement>) => {
+        if (handleValueChange) {
+            handleValueChange(INPUT_CHANGE, 'WORKER_NODE_COUNT', event.target.value, currentStep, errors);
+        }
+    };
+
+    const [selectedProvider, setSelectedProvider] = useState('CALICO');
+
+    const handleProviderChange = (event: ChangeEvent<HTMLSelectElement>) => {
+        setSelectedProvider(event.target.value);
+    };
+    return (
```

Comment:
I'd like to see this huge return broken up into smaller methods that lay out a small amount of HTML. I'm about to post a PR that does exactly that, if it's not clear what I'm getting at.